pkg/apis/core/types.go

```diff
@@ -3348,6 +3348,23 @@ type NodeSystemInfo struct {
 	Architecture string
 }
 
+// NodeConfigStatus describes the status of the config assigned by Node.Spec.ConfigSource
+type NodeConfigStatus struct {
+	// Assigned is the node's ack of the config source assigned via Node.Spec.ConfigSource,
+	// and is a direct reflection of that source.
+	// +optional
+	Assigned *NodeConfigSource
+	// Active reports the config source the node is actively using
+	// +optional
+	Active *NodeConfigSource
+	// LastKnownGood is the config source that will be used if an invalid config is assigned in the future
+	// +optional
+	LastKnownGood *NodeConfigSource
+	// Error describes any problems with the configuration
+	// +optional
+	Error string
```

Comment:
clarify whether the presence of Error means the Assigned config source will be ignored until changes are made to it

Reply:
will do